sycl/include/sycl/ext/intel/pipes.hpp

```diff
@@ -52,7 +52,7 @@ template <class _name, class _dataT, int32_t _min_capacity = 0> class pipe {
     (void)_Data;
     throw sycl::exception(
         sycl::make_error_code(sycl::errc::feature_not_supported),
-        "Pipes are not supported on a host device.");
+        "Device-side API are not supported on a host device. Please use host-side API instead..");
```

Comment:
I have a question, @steffenlarsen, do you know if it's possible to make this a compile time error instead of a runtime error when the user is trying to call a device-side API on the host?

Reply:
It is not possible as both paths are compiled inside a kernel. A `sycl::exception` is the best option here.